core/src/main/java/fr/sncf/osrd/standalone_sim/SafetySpeed.kt

```diff
@@ -0,0 +1,94 @@
+package fr.sncf.osrd.standalone_sim
+
+import fr.sncf.osrd.api.api_v2.standalone_sim.SimulationScheduleItem
+import fr.sncf.osrd.conflicts.TravelledPath
+import fr.sncf.osrd.railjson.schema.schedule.RJSTrainStop
+import fr.sncf.osrd.sim_infra.api.Path
+import fr.sncf.osrd.sim_infra.api.RawInfra
+import fr.sncf.osrd.sim_infra.api.Route
+import fr.sncf.osrd.sim_infra.api.ZonePath
+import fr.sncf.osrd.sim_infra.api.ZonePathId
+import fr.sncf.osrd.sim_infra.impl.ChunkPath
+import fr.sncf.osrd.utils.DistanceRangeMap
+import fr.sncf.osrd.utils.distanceRangeMapOf
+import fr.sncf.osrd.utils.indexing.StaticIdx
+import fr.sncf.osrd.utils.indexing.StaticIdxList
+import fr.sncf.osrd.utils.units.Offset
+import fr.sncf.osrd.utils.units.Speed
+import fr.sncf.osrd.utils.units.kilometersPerHour
+import fr.sncf.osrd.utils.units.meters
+
+/**
+ * Compute safety speed ranges, areas where the train has a lower speed limit because of a scheduled
+ * stop. For details, see https://osrd.fr/en/docs/reference/design-docs/timetable/#modifiable-fields
+ * (or google "VISA (VItesse Sécuritaire d'Approche)" for resources in French)
+ */
+fun makeSafetySpeedRanges(
+    infra: RawInfra,
+    chunkPath: ChunkPath,
+    routes: StaticIdxList<Route>,
+    schedule: List<SimulationScheduleItem>
+): DistanceRangeMap<Speed> {
+    val zonePaths = routes.flatMap { infra.getRoutePath(it) }
+    val zonePathStartOffset = getRoutePathStartOffset(infra, chunkPath, zonePaths)
+    val signalOffsets = getSignalOffsets(infra, zonePaths, zonePathStartOffset)
+
+    val stopsWithSafetySpeed =
+        schedule.filter {
+            it.stopFor != null && it.receptionSignal != RJSTrainStop.RJSReceptionSignal.OPEN
```

Comment:
2 suggestions here:
```suggestion
            it.receptionSignal.isStopOnClosedSignal()
```
1. There are checks that `receptionSignal` matches `stopFor` before that, so I would only protect for crashes, not for functionnality here (so no check on `stopFor`). Maybe raise an error before in core if not trusting the client?
2. `isStopOnClosedSignal()` is probably more future-proof if enum evolves.